shell/shared/grain.js

```diff
@@ -1148,52 +1312,68 @@ Router.map(function () {
       ];
     },
 
-    data: function() {
-      if (!this.ready || Meteor.loggingIn()) {
-        return;
-      }
-      if (Meteor.userId() && !Session.get("visit-token-" + this.params.token)) {
-        var tokenInfo = TokenInfo.findOne(this.params.token);
-        if (!tokenInfo || !tokenInfo.apiToken) {
-          this.state.set("error", "invalid authorization token");
-        } else {
-          this.state.set("error", undefined);
-          var apiToken = tokenInfo.apiToken;
-          if (!Grains.findOne({_id: apiToken.grainId, userId: Meteor.userId()}) &&
-              !ApiTokens.findOne({userId: apiToken.userId, "owner.user.userId": Meteor.userId()})) {
+    onRun: function () {
+      var token = this.params.token;
+      var path = this.params.path;
+      var query = this.params.query;
+      var hash = this.params.hash;
+
+      Tracker.autorun(function() {
```

Comment:
But mostly it's that last one - I want to automatically open the session as soon as the user answers the "reveal identity" question, or automatically as soon as data arrives that shows my identity is already revealed, or that I own the grain.